From cdf80e62e230dad25b676f2dd12d6f328304b507 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Mon, 8 Oct 2018 11:59:48 +0900
Subject: [PATCH 30/30] nsis: Improve some messages

Show a message while waiting the uninstall of old version, or while
executing install.exe or uninstal.exe.
---
 nsis/gvim.nsi             | 44 +++++++++++++++++++++++++++++++++++++++
 nsis/lang/dutch.nsi       |  9 ++++++++
 nsis/lang/english.nsi     |  9 ++++++++
 nsis/lang/german.nsi      |  9 ++++++++
 nsis/lang/italian.nsi     |  9 ++++++++
 nsis/lang/japanese.nsi    |  9 ++++++++
 nsis/lang/simpchinese.nsi |  9 ++++++++
 nsis/lang/tradchinese.nsi |  9 ++++++++
 8 files changed, 107 insertions(+)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index 1e93c733f..3a61a9652 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -183,6 +183,47 @@ ReserveFile ${VIMSRC}\installw32.exe
 ##########################################################
 # Functions
 
+# Update the message in the (un)install files page
+# Based on http://nsis.sourceforge.net/DetailUpdate
+!macro _DetailUpdate un
+Function ${un}DetailUpdate
+  Exch $R0
+  Push $R1
+  Push $R2
+  Push $R3
+
+  FindWindow $R2 `#32770` `` $HWNDPARENT
+  GetDlgItem $R1 $R2 1006
+  SendMessage $R1 ${WM_SETTEXT} 0 `STR:$R0`
+  GetDlgItem $R1 $R2 1016
+
+  System::Call *(&t${NSIS_MAX_STRLEN}R0)i.R2
+  System::Call *(i0,i0,i0,i0,i0,iR2,i${NSIS_MAX_STRLEN},i0,i0)i.R3
+
+  !define /ifndef LVM_GETITEMCOUNT 0x1004
+  !define /ifndef LVM_SETITEMTEXT 0x102E
+  SendMessage $R1 ${LVM_GETITEMCOUNT} 0 0 $R0
+  IntOp $R0 $R0 - 1
+  System::Call user32::SendMessage(iR1,i${LVM_SETITEMTEXT},iR0,iR3)
+
+  System::Free $R3
+  System::Free $R2
+
+  Pop $R3
+  Pop $R2
+  Pop $R1
+  Pop $R0
+FunctionEnd
+!macroend
+!insertmacro _DetailUpdate ""
+!insertmacro _DetailUpdate "un."
+!macro DetailUpdate un Text
+  Push `${Text}`
+  Call ${un}DetailUpdate
+!macroend
+!define DetailUpdate `!insertmacro DetailUpdate ""`
+!define un.DetailUpdate `!insertmacro DetailUpdate "un."`
+
 # Get parent directory
 # Share this function both on installer and uninstaller
 !macro GetParent un
@@ -285,6 +326,7 @@ Section "$(str_section_old_ver)" id_section_old_ver
 	# run the install program to check for already installed versions
 	SetOutPath $TEMP
 	File /oname=install.exe ${VIMSRC}\installw32.exe
+	${DetailUpdate} "$(str_msg_wait_uninst)"
 	${Do}
 	  nsExec::Exec "$TEMP\install.exe -uninstall-check"
 	  Pop $3
@@ -652,6 +694,7 @@ SectionEnd
 ##########################################################
 Section -call_install_exe
 	SetOutPath $0
+	${DetailUpdate} "$(str_msg_registering)"
 	nsExec::Exec "$0\install.exe $1"
 	Pop $3
 SectionEnd
@@ -883,6 +926,7 @@ Section "un.$(str_unsection_register)" id_unsection_register
 !endif
 
 	# delete the context menu entry and batch files
+	${un.DetailUpdate} "$(str_msg_unregistering)"
 	nsExec::Exec "$0\uninstal.exe -nsis"
 	Pop $3
 
diff --git a/nsis/lang/dutch.nsi b/nsis/lang/dutch.nsi
index 3628c3445..64f8cd851 100644
--- a/nsis/lang/dutch.nsi
+++ b/nsis/lang/dutch.nsi
@@ -221,6 +221,15 @@ LangString str_msg_rm_exe_fail   ${LANG_DUTCH} \
 LangString str_msg_rm_root_fail  ${LANG_DUTCH} \
     "WAARSCHUWING: Kan $\"$vim_install_root$\" niet verwijderen omdat het niet leeg is!"
 
+LangString str_msg_wait_uninst   ${LANG_DUTCH} \
+    "Waiting the old version to be uninstalled..."
+
+LangString str_msg_registering   ${LANG_DUTCH} \
+    "Registering..."
+
+LangString str_msg_unregistering ${LANG_DUTCH} \
+    "Unregistering..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
diff --git a/nsis/lang/english.nsi b/nsis/lang/english.nsi
index 7acbfe990..159e4ab85 100644
--- a/nsis/lang/english.nsi
+++ b/nsis/lang/english.nsi
@@ -230,6 +230,15 @@ LangString str_msg_rm_exe_fail   ${LANG_ENGLISH} \
 LangString str_msg_rm_root_fail  ${LANG_ENGLISH} \
     "WARNING: Cannot remove $\"$vim_install_root$\", it is not empty!"
 
+LangString str_msg_wait_uninst   ${LANG_ENGLISH} \
+    "Waiting the old version to be uninstalled..."
+
+LangString str_msg_registering   ${LANG_ENGLISH} \
+    "Registering..."
+
+LangString str_msg_unregistering ${LANG_ENGLISH} \
+    "Unregistering..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
diff --git a/nsis/lang/german.nsi b/nsis/lang/german.nsi
index a828d183a..ffa3897d8 100644
--- a/nsis/lang/german.nsi
+++ b/nsis/lang/german.nsi
@@ -221,6 +221,15 @@ LangString str_msg_rm_root_fail  ${LANG_GERMAN} \
     "Achtung: Kann Verzeichnis $\"$vim_install_root$\" nicht entfernen, \
      weil es nicht leer ist!"
 
+LangString str_msg_wait_uninst   ${LANG_GERMAN} \
+    "Waiting the old version to be uninstalled..."
+
+LangString str_msg_registering   ${LANG_GERMAN} \
+    "Registering..."
+
+LangString str_msg_unregistering ${LANG_GERMAN} \
+    "Unregistering..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
diff --git a/nsis/lang/italian.nsi b/nsis/lang/italian.nsi
index 7b63ba627..5aa10454f 100644
--- a/nsis/lang/italian.nsi
+++ b/nsis/lang/italian.nsi
@@ -230,6 +230,15 @@ LangString str_msg_rm_exe_fail   ${LANG_ITALIAN} \
 LangString str_msg_rm_root_fail  ${LANG_ITALIAN} \
     "AVVISO: Non posso cancellare $\"$vim_install_root$\", non è vuota!"
 
+LangString str_msg_wait_uninst   ${LANG_ITALIAN} \
+    "Waiting the old version to be uninstalled..."
+
+LangString str_msg_registering   ${LANG_ITALIAN} \
+    "Registering..."
+
+LangString str_msg_unregistering ${LANG_ITALIAN} \
+    "Unregistering..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
diff --git a/nsis/lang/japanese.nsi b/nsis/lang/japanese.nsi
index 0393aa757..67078a0df 100644
--- a/nsis/lang/japanese.nsi
+++ b/nsis/lang/japanese.nsi
@@ -223,6 +223,15 @@ LangString str_msg_rm_exe_fail   ${LANG_JAPANESE} \
 LangString str_msg_rm_root_fail  ${LANG_JAPANESE} \
     "WARNING: Cannot remove $\"$vim_install_root$\", it is not empty!"
 
+LangString str_msg_wait_uninst   ${LANG_JAPANESE} \
+    "古いバージョンがアンインストールされるのを待っています..."
+
+LangString str_msg_registering   ${LANG_JAPANESE} \
+    "登録中..."
+
+LangString str_msg_unregistering ${LANG_JAPANESE} \
+    "登録解除中..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
diff --git a/nsis/lang/simpchinese.nsi b/nsis/lang/simpchinese.nsi
index 38b98df6a..a4e2c990a 100644
--- a/nsis/lang/simpchinese.nsi
+++ b/nsis/lang/simpchinese.nsi
@@ -217,6 +217,15 @@ LangString str_msg_rm_root_fail  ${LANG_SIMPCHINESE} \
     "警告：无法删除 Vim 安装目录“$vim_install_root”，\
      该目录下仍有其他文件。"
 
+LangString str_msg_wait_uninst   ${LANG_SIMPCHINESE} \
+    "Waiting the old version to be uninstalled..."
+
+LangString str_msg_registering   ${LANG_SIMPCHINESE} \
+    "Registering..."
+
+LangString str_msg_unregistering ${LANG_SIMPCHINESE} \
+    "Unregistering..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
diff --git a/nsis/lang/tradchinese.nsi b/nsis/lang/tradchinese.nsi
index b5a4c7c8a..63eea8438 100644
--- a/nsis/lang/tradchinese.nsi
+++ b/nsis/lang/tradchinese.nsi
@@ -218,6 +218,15 @@ LangString str_msg_rm_root_fail  ${LANG_TRADCHINESE} \
     "警告：無法刪除 Vim 安裝資料夾「$vim_install_root」，\
      該資料夾下仍有其他檔案。"
 
+LangString str_msg_wait_uninst   ${LANG_TRADCHINESE} \
+    "Waiting the old version to be uninstalled..."
+
+LangString str_msg_registering   ${LANG_TRADCHINESE} \
+    "Registering..."
+
+LangString str_msg_unregistering ${LANG_TRADCHINESE} \
+    "Unregistering..."
+
 
 ##############################################################################
 # Dialog Box                                                              {{{1
-- 
2.17.0

